src/org/mozilla/javascript/NativeMap.java

```diff
@@ -191,33 +187,20 @@ static void loadFromIterable(Context cx, Scriptable scope, ScriptableObject map,
         // Find the "add" function of our own prototype, since it might have
         // been replaced. Since we're not fully constructed yet, create a dummy instance
         // so that we can get our own prototype.
-        ScriptableObject dummy = ensureScriptableObject(cx.newObject(scope, map.getClassName()));
-        final Callable set =
-                ScriptRuntime.getPropFunctionAndThis(dummy.getPrototype(), "set", cx, scope);
+        Scriptable proto = ScriptableObject.getClassPrototype(scope, map.getClassName());
```

Comment:
These are the only code change in this class.
I have changed this to `getClassPrototype` instead of creating a dummy object and I reuse the code in 
`ScriptRuntime.loadFromIterable`